alpha/veneer/composite/composite.go

```diff
@@ -0,0 +1,48 @@
+package composite
+
+import (
+	"context"
+	"fmt"
+)
+
+type BuilderMap map[string]Builder
+
+type CatalogBuilderMap map[string]BuilderMap
+
+type Veneer struct {
+	CatalogBuilders CatalogBuilderMap
+}
+
+// TODO(everettraven): do we need the context here? If so, how should it be used?
+func (v *Veneer) Render(ctx context.Context, config *CompositeConfig, validate bool) error {
+	// TODO(everettraven): should we return aggregated errors?
+	for _, component := range config.Components {
+		if builderMap, ok := v.CatalogBuilders[component.Name]; ok {
+			if builder, ok := builderMap[component.Strategy.Veneer.Schema]; ok {
+				// run the builder corresponding to the schema
+				err := builder.Build(component.Destination.Path, component.Strategy.Veneer)
+				if err != nil {
+					return fmt.Errorf("building component %q: %w", component.Name, err)
+				}
+
+				if validate {
+					// run the validation for the builder
+					err = builder.Validate(component.Destination.Path)
+					if err != nil {
+						return fmt.Errorf("validating component %q: %w", component.Name, err)
+					}
+				}
+				// TODO(everettraven): Should we remove the built FBC if validation fails?
```

Comment:
If we presume that the failure reporting is useful, then IMHO don't delete the erroneous FBC.  We don't error if they overwrite it.